aten/src/ATen/templates/Functions.cpp

```diff
@@ -25,4 +25,8 @@ std::tuple<Tensor,Tensor> std_mean(const Tensor& self, int dim) {
 
 ${function_definitions}
 
-}
+namespace redispatch {
+    ${function_redispatch_definitions}
+} // namespace redispatch
```

Comment:
This causes the size of `Functions.cpp` to double from ~18k to ~36k lines. Is it worth moving to a separate file like `RedispatchFunctions.cpp`? I know we have some logic in codegen for sharing big files to prevent a massive slowdown to build times, not sure if this is at that level or not.

Actually, maybe it's worth it. `TraceTypeEverything.cpp` is ~59k lines, and its sharded files are each ~12k lines.